languages/go/exercises/concept/basic-slices/.meta/design.md

```diff
@@ -0,0 +1,41 @@
+## Goal
+
+The goal of this exercise is to teach the student the basics on how to work with slices in Go.
+
+## Things to teach
+
+For example:
+
+- instantiate (len, cap)
+- append
+- set value
+- read value
+- remove value
+- taking length
+- subslices
+- expanding slices (e.g. in `append(slice1, slice2...)`)
+
+## Things not to teach
+
+For example:
+
+- inner workings of slices
```

Comment:
Doesn't the introduction document describe the inner working?

Reply:
Yes, they are briefly mentioned but there are no practical teachings on that in the code. For example what happens if I take a subslice of a slice and then change the original/subslice to the other slice? What happens if I append to a slice but the underlying array has no more space? In what way is the append function not `pure`? What does it mean practically?

Reply:
@ErikSchierboom: Would you remove the part on "based on arrays" and that they are technically "pointers to an array with a length" from the introduction then?

Reply:
Or maybe rename this bullet point to: 
```suggestion
- deep dive into the inner workings of slices
```

Reply:
I like the suggestion! If we apply that, I think the introduction is fine.